tests/adapters/servers/test_tcp.py

```diff
@@ -0,0 +1,90 @@
+from asyncio import StreamReader, StreamWriter
+from typing import AsyncIterable, Callable
+
+import pytest
+from mock import ANY, MagicMock, Mock, patch
+from mock.mock import AsyncMock, create_autospec
+
+from tickit.adapters.servers.tcp import TcpServer
+
+
+@pytest.fixture
+def tcp_server() -> TcpServer:
+    return TcpServer()
+
+
+@pytest.fixture
+def on_connect() -> Callable[[], AsyncIterable[bytes]]:
+    async def on_connect():
+        yield b"hello"
+
+    return on_connect
+
+
+@pytest.fixture
+def handler() -> Callable[[bytes], AsyncIterable[bytes]]:
+    async def _handler(message: bytes) -> Callable[[], AsyncIterable[bytes]]:
+        async def async_iterable() -> AsyncIterable[bytes]:
+            yield b"hello"
+            yield b""
+            yield b"blue"
+            yield b"world"
+            yield b""
+
+        return async_iterable
+
+    return _handler  # type: ignore
```

Comment:
MyPy thinks the type of `_handler` is: `Callable[[bytes], Coroutine[Any, Any, Callable[[], AsyncIterable[bytest]]]]`.

Reply:
I'm still fairly sure mypy is correct, though will tag @callumforrester for his take

Reply:
I don't get a mypy error from this test...
That said, I think `_handler` is of type:

```python
Callable[[bytes], Awaitable[AsyncIterable[bytes]]]
```

Reply:
`Awaitable[X]` is a more general form of `Coroutine[Any, Any, X]`

Reply:
```
tests/adapters/servers/test_tcp.py:36: error: Incompatible return value type (got "Callable[[bytes], Coroutine[Any, Any, Callable[[], AsyncIterable[bytes]]]]", expected "Callable[[bytes], Awaitable[AsyncIterable[bytes]]]")
```

Reply:
This seems to work, though I don't know if I need the nested function calls or not:
```
@pytest.fixture
def handler() -> Callable[[bytes], Awaitable[AsyncIterable[bytes]]]:
    async def _handler(message: bytes) -> AsyncIterable[bytes]:
        async def async_iterable() -> AsyncIterable[bytes]:
            yield b"hello"
            yield b""
            yield b"blue"
            yield b"world"
            yield b""

        return async_iterable()

    return _handler
```

Reply:
Best way to tell is to intentionally break the test

Reply:
Seems like it can't be `Awaitable` with a single function, I suppose that makes sense if `Awaitable` means, "returned by an `async` function". Which I think it does?